applications/asset_tracker_v2/src/main.c

```diff
@@ -432,37 +380,19 @@ static void data_get(void)
 		app_module_event->data_list[count++] = APP_DATA_MODEM_STATIC;
 	}
 
-	if (!app_cfg.no_data.neighbor_cell) {
-		app_module_event->data_list[count++] = APP_DATA_NEIGHBOR_CELLS;
-		app_module_event->timeout = DATA_FETCH_TIMEOUT_NEIGHBORHOOD_SEARCH;
-	}
-
-	/* The reason for having at least 75 seconds sample timeout when
-	 * requesting GNSS data is that the GNSS module on the nRF9160 modem will always
-	 * search for at least 60 seconds for the first position fix after a reboot. This limit
-	 * is enforced in order to give the modem time to perform scheduled downloads of
-	 * A-GPS data from the GNSS satellites.
-	 *
-	 * However, if A-GPS data has been downloaded via the cloud connection and processed
-	 * before the initial GNSS search, the actual GNSS timeout set by the application is used.
-	 *
-	 * Processing A-GPS before requesting GNSS data is enabled by default,
-	 * and the time that the application will wait for A-GPS data before including GNSS
-	 * in sample requests can be adjusted via the
-	 * CONFIG_APP_REQUEST_GNSS_WAIT_FOR_AGPS_THRESHOLD_SEC Kconfig option. If this option is
-	 * set to -1, the application module will not request GNSS unless A-GPS data has been
-	 * processed.
-	 *
-	 * If A-GPS data has been processed the sample timeout can be ignored as the
-	 * GNSS will most likely time out before the sample timeout expires.
-	 *
-	 * When GNSS is requested, set the sample timeout to (GNSS timeout + 15 seconds)
-	 * to let the GNSS module finish ongoing searches before data is sent to cloud.
-	 */
+	if (IS_ENABLED(CONFIG_DATA_SAMPLE_GNSS_DEFAULT) ||
+	    IS_ENABLED(CONFIG_DATA_SAMPLE_NEIGHBOR_CELLS_DEFAULT)) {
+		app_module_event->data_list[count++] = APP_DATA_LOCATION;
 
-	if (request_gnss() && !app_cfg.no_data.gnss) {
-		app_module_event->data_list[count++] = APP_DATA_GNSS;
-		app_module_event->timeout = MAX(app_cfg.gnss_timeout + 15, 75);
+		/* Set application module timeout */
+		app_module_event->timeout = 120;
+		if (app_cfg.active_wait_timeout <= 3) {
+			app_module_event->timeout = app_cfg.active_wait_timeout;
+		} else if (app_cfg.active_wait_timeout < 20) {
+			app_module_event->timeout = app_cfg.active_wait_timeout - 2;
+		} else if (app_cfg.active_wait_timeout < 130) {
+			app_module_event->timeout = app_cfg.active_wait_timeout - 10;
+		}
```

Comment:
This is what I came up with at the moment. I thought two minutes would be kind of default data publishing interval safe guard if all modules haven't responded.
If `active_wait_time` gets shorter and especially below one minute, GNSS stops working based on network configuration as LTE will block GNSS radio. But as said, this is very network dependent.

If ATv2 is designed to be used with small `active_wait_timeout` such as 10s, I don't know how this is supposed to work. Comments are welcome.

Reply:
You have also two other configurations that affects this. In passive mode, `movement_resolution` is used to limit how often data is sent. If the device is constantly moving, data will be sampled every `movement_resolution`. `movement_timeout` acts the same way as `active_wait_timeout` and is also used in passive mode.

I wonder if it makes sense to just use a set timeout of 120 seconds to avoid accommodating for all the different timer values (and modes). If the user sets too low of a sample interval, the application will simply send data as fast as it can. This can be documented.

Reply:
True, `movement_timeout` should have been considered too. I'm more than happy to simplify this and set hard coded 120s. Or maybe 110s as `active_wait_timeout` default is 120s.
The only problem, which is already in main branch ATv2, I see is that if you set `active_wait_timeout` to 20s, data is sent after `gnss_timeout` dependent value which by default is 75s (in the future this would be `location_timeout` and longer, such as 5mins etc by default). I mean BATTERY etc data won't be sent either. This is what I observed when testing with `active_wait_timeout` timeout values.
And the problem is that there are no warnings etc. when misconfigurations take place.

Reply:
Yes, but this is only expected to occur if obtaining location takes a lot of time. With A-GPS, 20 seconds might be enough to be able to respect the 20 second sample interval. How often from a general point of view will the location API be able to provide location? I think we should rather optimize for longer sample request intervals rather than short if we are to choose. We could add documentation that states that this might be an issue for lower intervals.

Reply:
FYI.
If we now remove` gnss_timeout` and add `location_timeout`, it means that we will have default value for `gnss_timeout` which is 120s. If location timeout is set by the user to 60s, 90s or 120s, fallback to cellular will never occur if GNSS fix is not acquired. So, `location_timeout` should really be set to some fairly large value such as 300s etc.

Reply:
Based on yesterday's discussions, app timeout is set as follows:
```
app_module_event->timeout = (app_cfg.active_mode) ?
            MIN(app_cfg.active_wait_timeout - 5, 110) :
            MIN(app_cfg.movement_resolution - 5, 110);
app_module_event->timeout = MAX(app_module_event->timeout, 5);
```